src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionPerfTests.cs

```diff
@@ -780,5 +781,200 @@ public void DefiniteAssignment_ManySwitchCasesAndLabels()
+            Assert.Equal("o.F( ", expr.ToString());
+            _ = model.GetTypeInfo(expr);
+        }
+
+        [Fact]
+        public void ExtensionOverloadsDistinctClasses_04()
+        {
+            // public abstract class A
+            // {
+            //     public static void F1(this object obj) { }
+            //     public static void F3(this object obj) { }
+            // }
+            // public abstract class B : A
+            // {
+            //     public static void F2(this object obj) { }
+            //     public static void F3(this object obj) { }
+            // }
+            string sourceA = """
+                .assembly extern mscorlib { .ver 4:0:0:0 .publickeytoken = (B7 7A 5C 56 19 34 E0 89) }
+                .assembly extern System.Core { }
+                .assembly '<<GeneratedFileName>>'
+                {
+                    .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                }
+                .class public abstract A
+                {
+                  .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                  .method public hidebysig specialname rtspecialname instance void .ctor() { ret }
+                  .method public static void F1(object o)
+                  {
+                    .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                    ret
+                  }
+                  .method public static void F3(object o)
+                  {
+                    .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                    ret
+                  }
+                }
+                .class public abstract B extends A
+                {
+                  .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                  .method public hidebysig specialname rtspecialname instance void .ctor() { ret }
+                  .method public static void F2(object o)
+                  {
+                    .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                    ret
+                  }
+                  .method public static void F3(object o)
+                  {
+                    .custom instance void [System.Core]System.Runtime.CompilerServices.ExtensionAttribute::.ctor() = ( 01 00 00 00 )
+                    ret
+                  }
+                }
+                """;
+            var refA = CompileIL(sourceA, prependDefaultHeader: false);
+
+            // The calls to B.F3(o) and o.F3() should bind to B.F3 and should not be
+            // considered ambiguous with A.F3 because B is derived from A.
```

Comment:
Does the language specify this?

Reply:
I'm not sure if the language specifies this, but this is the existing behavior so I wanted to avoid an unnecessary breaking change.